src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -324,12 +340,76 @@ private void PossiblyGlobArg(string arg, bool usedQuotes)
             }
 #endif // UNIX
 
+            // Check if the start of the path is a filesystem drive
+            if (stringConstantType != StringConstantType.SingleQuoted)
+            {
+                arg = ResolvePath(arg, Context);
+            }
+
             if (!argExpanded)
             {
                 _arguments.Append(arg);
             }
         }
 
+        /// <summary>
+        /// Check if string is prefixed by psdrive, if so, expand it if filesystem path.
+        /// </summary>
+        /// <param name="path">The potential PSPath to resolve.</param>
+        /// <param name="context">The current ExecutionContext.</param>
+        /// <returns>Resolved PSPath if applicable otherwise the original path</returns>
+        internal static string ResolvePath(string path, ExecutionContext context)
+        {
+            if (ExperimentalFeature.IsEnabled("PSNativePSPathResolution"))
+            {
+                string driveName;
+#if !UNIX
+                if (path.Equals("~") || path.StartsWith("~" + Path.DirectorySeparatorChar) || path.StartsWith("~" + Path.AltDirectorySeparatorChar))
```

Comment:
Our pattern is to use explicit StringComparison type.
Below too.

Reply:
Will update